site/content/integrate/apps/interactivity/dynamic-form-example/_index.md

```diff
@@ -0,0 +1,544 @@
+- `/forms/create-ticket-post-menu/submit`
+
+### Opening a modal
+
+Now let's process the post menu item click, and return an `AppForm` to show as a modal:
+
+```ts
+app.post('/forms/create-ticket-post-menu/submit', (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Extract the Post from the expanded context
+    const post: Post = call.context.post;
+
+    const call: AppCall = {
+        path: '/forms/create-ticket',
+        expand: {
+            acting_user: 'summary', // Expand the acting user so we can get the user's username for our post
+            team: 'summary', // Expand the current team so we can construct a permalink for the selected post using the team's name
+        },
+    };
+
+    const form: AppForm = {
+        title: 'Create a Ticket',
+        fields: [
+            {
+                name: 'summary',
+                value: '',
+                modal_label: 'Summary',
+                type: 'text',
+                is_required: true, // This makes it so the user must provide this value before submitting the form
+                min_length: 10, // We set a minimum length of 10 characters for this field
+            },
+            {
+                name: 'description',
+                value: post.message, // Use the Post's message as the default value for the description field
+                modal_label: 'Description',
+                type: 'text',
+                subtype: 'textarea',
+            },
+        ],
+    };
+
+    const callResponse: AppCallResponse = {
+        type: 'form',
+        call,
+        form,
+    }
+
+    res.json(callResponse);
+});
+```
+
+The user is shown a form with text fields labeled "Summary" and "Description". The Description field is prepopulated with the `Post`'s message.
+
+By defining the form's `Call` with the path `/forms/create-ticket`, we've implicitly registered three endpoints for this form to use:
+- `/forms/create-ticket/lookup` is used to fetch autocomplete values
+- `/forms/create-ticket/form` is used to dynamically redefine the modal when a particular field's value is chosen
+- `/forms/create-ticket/submit` is used to process the form's submission
+
+### Processing the form submission
+
+Now let's process the form submitted at the path `/forms/create-ticket/submit`
+
+We've specified in the `expand` clause to include the acting `User` that is submitting the form, as well as the `Team` so we can construct a post's permalink using the team's name.
+
+```ts
+import Client4 from 'mattermost-redux/client/client4';
+
+type CreateTicketFormSubmission = {
+    summary: string;
+    description: string;
+};
+
+app.post('/forms/create-ticket/submit', async (req, res) => {
+    const call: AppCallRequest = req.body;
+
+    // Extract the submitted form values
+    const {summary, description}: CreateTicketFormSubmission = call.values;
```

Comment:
2/5 add some mention of call.values outside of the code and under `Processing the form submission`.  Let the use know that `call.values` will contain the forms previous values.